shared/anachronism_detector.py

```
@@ -0,0 +1,355 @@
+                                for issue in issues:
+                                    issue["sample_num"] = line_num
+                                    issue["source"] = sample.get("source", "unknown")
+                                    results["issues_by_type"][issue["type"]] += 1
+                                file_issues.extend(issues)
+
+                if file_issues:
+                    results["files_with_issues"] += 1
+                    results["issues_by_file"][zst_file.name] = file_issues
+                    results["total_issues"] += len(file_issues)
+
+            except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)